ush/om_data.py

```diff
@@ -0,0 +1,390 @@
+        self.logger.info(f'   dir_struct: {self.dir_struct}')
+        self.logger.info('============' + '\n')
+
+# --------------------------------------------------------------------------------------------
+
+    def set_data_types(self, template):
+        """
+        Set data_type and data_subtype using the plot template name.  Note
+        that 'con' and 'ozn' types have some special cases for subtype.
+
+        Paremeters:
+            template (str): Name of plot template file
+        Return:
+            None
+        """
+
+        self.data_type = template[:3].lower()
+        stype = template[3:].lower()
+
+        self.data_subtype = stype
+
+        match self.data_type:
+            case 'con':
+                if stype in ['time', 'vert']:
+                    self.data_subtype = 'time_vert'
+                elif stype in ['hist', 'horz']:
+                    self.data_subtype = 'horz_hist'
+
+            case 'ozn':
+                if stype == 'summary':
+                    self.data_subtype = 'time'
+
+# --------------------------------------------------------------------------------------------
+
+    def load_data_files(self, yaml_file):
+
+        """
+        Load control and data file names from plot_dict into OM_data class object.
+
+        Parameters:
+            yaml_file (file): yaml file with plot dict info
+        Return:
+            None
+        """
+
+        try:
+            with open(yaml_file, 'r') as yaml_file_opened:
+                pd = yaml.safe_load(yaml_file_opened)
+
+                for d in pd.get('datasets'):
+                    if 'control_file' in d.keys():
+                        self.ctl_file.append(os.path.basename(d['control_file'][0]))
+                    self.data_files.append([os.path.basename(f) for f in d['filenames']])
+
+        except Exception as e:
+            self.logger.info('Warning: unable to load yaml file into dict, ' +
+                             f'errors when attempting to load: {yaml_file}, error: {e}')
+
+# --------------------------------------------------------------------------------------------
+
+    def set_dir_struct(self):
+        """
+        Determine which data directory structure is in use. Supported structures are:
+            obs-mon:   used for development, data is simply in 'ozn_data', 'rad_data', etc.
+                       and is not gzipped or tarred.
+            glb-wkflw: global workflow structure
+            gfs-ops:   operational gfs structure
+            leg-mon:   legacy DA monitor structure
+
+        Parameters:
+            None
+        Return:
+            None
+        """
+
+        mon = self.data_type + 'mon'
+        om_test = os.path.join(self.data_src, self.data_type + '_data')
+        gw_test = ""
+        go_test = ""
+        lm_test = ""
+
+        match mon:
+            case 'oznmon':
+                gw_test = os.path.join(self.data_src, 'gdas.' + self.pdate[:-2], self.pdate[-2:],
```

Comment:
since those slices are used a lot, would it be beneficial to define `gdas.YYYYMMDD/HH` as a variable?

Reply:
also, should we not hard code gdas for future use by other systems?